src/routes/blog/ide-ux-bof-scipy-2024/+page.md

```diff
@@ -0,0 +1,141 @@
+
+
+## GPU compute integration
+
+One fairly specific topic that got a disproportionate amount of attention was incorporating better support for GPU compute (e.g. CUDA) within IDEs and interfaces, thanks to Nvidia representatives in the room posing the initial question.
+They stated that Nvidia was "getting serious" about Python and CUDA, and they would love to "hear people's gripes about developing GPU apps in Python, what tooling you use and what problems you face and how do you debug them".
+They also asked for people's experiences working in Python together with native code (C++).
+
+Responding first to the question on native code, one participant described how they had to compile a Fortran module and import it from Python.
+They just used `f2py` and the workflow was already in there and worked: they'd change the native code, recompile and rinse and repeat.
+However, they said it Would be nice to offer a workflow to shorten that development loop.
+It was mentioned that this was particularly a problem with two separate codebases, where changes in one can affect the other, which is difficult to conceptualize and work with the coupling between them.
+The Nvidia representatives then asked them how they would like to have a debugger with the ability to step into their code from Python to Fortran but still able to inspect your variables on the Python side, to which they replied "Sounds awesome!".
+
+Another attendee has a postdoc using a bunch of A100 GPUs, and asked them whether they were getting the most performance out of it.
+The postdoc's reply, which the attendee found rather telling, was that "PyTorch runs"—it was painful enough for them to just get it running. so they were not particularly interested in spending even more time optimizing it further.
+The participant now has new students doing JAX and expect the same thing to happen, since diving into hardware level optimization is something that students really won't do and can be very intimidating.
+They have students who don't know how to use the 4 CPUs in their laptop, much less the GPUs in the supercomputer center.
+When asked what they feel is the main problem causing this, such as documentation or tooling, the researcher replied it was courage on the student's part to actually "go down there" and dig into the low-level code.
+The postdoc clearly likes the platform and bought 40 new A100s, and is fine with the platform abstracting that for him.
+But, they said, it poses a real sociological challenge to get students used to using, much less optimizing GPUs.
+
+Finally, it was asked what IDEs could do to make GPU workflows more convenient.
+In reply, participants suggested better integration with NVSMI or an integrated dashboard of GPU usage, which would allow users to see that, for instance, a particular kernel is crunching data is too big to use the full potential of the GPU.
```

Comment:
```suggestion
In reply, participants suggested better integration with NVSMI or an integrated dashboard of GPU usage, which would allow users to see that, for instance, a particular kernel crunching data is too big to use the full potential of the GPU.
```

Seems unnecessary (otherwise I don't understand the meaning of this sentence).